data_juicer/ops/mapper/generate_caption_mapper.py

```diff
@@ -0,0 +1,157 @@
+        self.model_in_ctx = None
+        self.img_processor_in_ctx = None
+        self.caption_num = caption_num
+        self.keep_candidate_mode = keep_candidate_mode
+        self.extra_args = kwargs
+
+    def compute_stats(self, sample, context=True):
+        # there is no image in this sample
+        if self.image_key not in sample or not sample[self.image_key]:
+            sample[Fields.stats][
+                StatsKeys.image_text_matching_score] = np.array(
+                    [], dtype=np.float64)
+            return sample
+
+        # load images for context re-using
+        loaded_image_keys = sample[self.image_key]
+        images = {}
+        for loaded_image_key in loaded_image_keys:
+            if context and loaded_image_key in sample[Fields.context]:
+                # load from context
+                images[loaded_image_key] = sample[
+                    Fields.context][loaded_image_key]
+            else:
+                if loaded_image_key not in images:
+                    # avoid load the same images
+                    image = load_image(loaded_image_key)
+                    images[loaded_image_key] = image
+                    if context:
+                        # store the image data into context
+                        sample[Fields.context][loaded_image_key] = image
+
+        # load model and processor for context re-using
+        if context and self.model_in_ctx is None:
+            model, img_processor = get_model(model_key=self.model_key,
+                                             usage='conditional_generation')
+            self.model_in_ctx = model
+            self.img_processor_in_ctx = img_processor
+
+        return sample
+
+    def process(self, sample):
+        # there is no image in this sample
+        if self.image_key not in sample or not sample[self.image_key]:
+            sample[Fields.stats][StatsKeys.image_sizes] = np.array(
+                [], dtype=np.float64)
+            return sample
+
+        # 1. load image(s)
+        loaded_image_keys = sample[self.image_key]
+        images = {}
+        for loaded_image_key in loaded_image_keys:
+            assert loaded_image_key in sample[Fields.context], \
+                "Image should has been loaded in 'compute_stats' calling"
+            # load from context
+            images[loaded_image_key] = sample[Fields.context][loaded_image_key]
+
+        # 2. generate candidate caption(s) in batch manner
+        generated_text_candidates = []
+        for n in range(self.caption_num):
+            inputs = self.img_processor_in_ctx(images=images.values(), )
+            generated_ids = self.model_in_ctx.generate(**inputs)
+            generated_text = self.img_processor_in_ctx.batch_decode(
+                generated_ids, skip_special_tokens=True)
+            generated_text_candidates.append(' '.join(generated_text))
+
+        # 3. reduce the captions according to given mode
```

Comment:
Replacing text of sample directly may be dangerous, specailly when SpecialToken in text, such as `EOC`.